File: csharp/Gherkin/Ast/Location.cs
using System;
using System.Linq;

namespace Gherkin.Ast
{
	public class Location
	{
		public string FilePath { get; private set; }
		public int Line { get; private set; }
		public int Column { get; private set; }

		public Location(string filePath, int line = 0, int column = 0)
		{
			FilePath = filePath;
			Line = line;
			Column = column;
		}
	}
}

File: java/src/main/java/gherkin/Location.java
package gherkin;

public class Location {
    private final int lineNumber;
    private int column;

    public Location(int lineNumber) {
        this.lineNumber = lineNumber;
    }

    public void setColumn(int column) {
        this.column = column;
    }

    @Override
    public String toString() {
        return lineNumber + ":" + column;
    }
}
